tests/test_exception.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from wait_for import wait_for, TimedOutError
+
+
+class MyError(Exception):
+    """A sample exception for use by the tests in this module."""
+
+
+def raise_my_error():
+    """Raise ``MyError``."""
+    raise MyError()
+
+
+def test_handle_exception_v1():
+    """Don't set ``handle_exception``.
+
+    An exception raised by the waited-upon function should bubble up.
+    """
+    with pytest.raises(MyError):
+        wait_for(raise_my_error)
+
+
+def test_handle_exception_v2():
+    """Set ``handle_exception`` to false.
+
+    An exception raised by the waited-upon function should bubble up.
+    """
+    with pytest.raises(MyError):
+        wait_for(raise_my_error, handle_exception=False)
+
+
+def test_handle_exception_v3():
```

Comment:
Is this test valid? Or is `wait_for()` supposed to catch the exception, compare it to `fail_condition` and discover a mismatch (b/c `e != False`), and therefore return a `WaitForResult(e, ...)`?

Reply:
@Ichimonji10 sorry for the delay in review - correct, that is the intended behavior of wait_for here. The exception is suppressed, and the loop continues looking for fail_condition to trigger a 'normal' return or TimedOutError.